src/hotp/mod.rs

```diff
@@ -1,3 +1,85 @@
-pub fn generate() {}
+/// Default layer to generate a HOTP
+/// # Arguments
+///
+/// * `digest` - Byte digest generated by hashing the secret key
+///
+pub fn generate_default(
+    digest: Vec<u8>
+) -> String {
+    generate(None, digest)
+}
+
+
+/// Generates a HOTP
+///
+/// # Arguments
+///
+/// * `digits` - Optional parameter that defines the length of the one time pad (default: 6)
+/// * `digest` - Byte digest generated by hashing the secret key
+///
+pub fn generate(
+    digits: Option<u32>,
+    digest:  Vec<u8>, // Assume the digest is generated correctly. I don't like the cyclical nature that speakeasy has generateHotp -> digest -> generateHotp. Thoughts?
+) -> String {
+    let defined_digits = match digits {
+        Some(d) => d,
+        None => 6
+    };
+
+    let offset = match digest.last() {
+        Some(o) => o & 0xf,
+        None => 0 // 0 & 0xf == 0
+    };
+
+    /*
+    This section works to fill up the unsigned 32 bit number by:

```

Comment:
can you go ahead and add the comment to the spec for this? just for easy traceability